src/drivers/imx/interrupt.c

```diff
@@ -155,10 +167,36 @@ static void irqstr_unmask_int(uint32_t irq)
 
 	if (irq < IRQSTR_RESERVED_IRQS_NUM || irq >= IRQSTR_IRQS_NUM)
 		return; // Unusable interrupts
+
+	irq = irqstr_fixup_irq(irq);
+
 	mask = IRQSTR_INT_MASK(irq);
 	irqstr_update_bits(IRQSTR_CH_MASK(IRQSTR_INT_REG(irq)), mask, mask);
 }
 
+#ifdef CONFIG_IMX8M
+
+/* Quirk of the driver in SOF (Quirk is specific to 8MP):
+ * -> IRQSTR has 5 input channels each with 32 interrupts
+ * -> IRQSTR has 3 output channels each with 64 interrupts
+ * -> IRQ in[31:0]    => IRQ out[63:0]   (output channel #0)
+ * -> IRQ in[32:63]   => IRQ out[95:32]  (output channel #1, low half)
```

Comment:
I think you should still keep 32-bit intervals on the right for clarity. Or perhaps say that 32:95 go to 64:127 or something. These comments are rather confusing from what I can say.

Reply:
@paulstelian97 you are right on this, will update.